tests/neptune/alpha/test_client.py

```diff
@@ -87,6 +87,44 @@ def test_resume(self):
         self.assertEqual(exp._uuid, uuid.UUID('12345678-1234-5678-1234-567812345678'))
         self.assertIsInstance(exp.get_structure()["test"], String)
 
+    @patch("neptune.alpha.internal.init_impl.sys.argv", ["main.py"])
+    @patch("neptune.alpha.internal.init_impl.os.path.isfile", new=lambda file: "." in file)
+    @patch('neptune.alpha.internal.utils.glob', new=lambda path: [path.replace('*', 'file.txt')])
+    @patch('neptune.alpha.internal.utils.os.path.abspath',
+           new=lambda path: os.path.normpath("/home/user/main_dir/" + path))
+    @patch('neptune.alpha.internal.utils.os.getcwd', new=lambda: "/home/user/main_dir")
+    def test_entrypoint(self):
+        exp = init(connection_mode='debug')
+        self.assertEqual(exp["source_code/entrypoint"].get(), "main.py")
+
+        exp = init(connection_mode='debug', source_files=[])
+        self.assertEqual(exp["source_code/entrypoint"].get(), "main.py")
+
+        exp = init(connection_mode='debug', source_files=["../*"])
+        self.assertEqual(exp["source_code/entrypoint"].get(), "main_dir/main.py")
+
+        exp = init(connection_mode='debug', source_files=["internal/*"])
+        self.assertEqual(exp["source_code/entrypoint"].get(), "main.py")
```

Comment:
Again according to my understanding entrypoint should be `../main.py`.

Reply:
It is not because there is a requirement that common root cannot be nested inside the working directory. If common root is calculated to be such, it is fixed to match the working directory. Because of that, effective entrypoint is still "main.py"